### PR TITLE
Add 802.1x support on the fortigate

### PR DESCRIPTION
**Pull request by lzammit**
# Description
Add 802.1x support on the fortigate 50E running fortiOS 5.4

# Impacts
Fortigate switches
Documentation

# Delete branch after merge
YES 

# NEWS file entries
802.1x support on the Fortigate 50E

### CODE REVIEW

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 1]
```diff
@@ -4890,6 +4890,49 @@ Secure SSID
 
 The final step is to enable SNMP Agent and SNMP Traps on the controller. This is done under the following tab in the administrative UI: *Advanced -> SNMP*.
 
+Fortigate
```
Comment (extrafu) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: Fortigate -> Fortinet FortiGate

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 2]
```diff
@@ -4890,6 +4890,49 @@ Secure SSID
...
+~~~~~~~~~
+
+This section shows how to configure a 802.1x SSID on a Fortigate 50E. The 802.1x was test on a FortiOS 5.4.
```
Comment (extrafu) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: idem.
802.1X

Comment (extrafu) [reply docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: was tested

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 3]
```diff
@@ -4890,6 +4890,49 @@ Secure SSID
...
+
+You will need to have the CLI access on the Fortigate to do the configuration.
```
Comment (extrafu) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: idem

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 4]
```diff
@@ -4890,6 +4890,49 @@ Secure SSID
...
+ FGT50E (packetfence) # config accounting-server 
+ FGT50E (accounting-server) # edit 1
+ new entry '1' added
+ FGT50E (1) # set status enable
+ FGT50E (1) # set server 192.168.1.5
+ FGT50E (1) # set secret useStrongerSecret
+ FGT50E (1) # end
+ FGT50E (packetfence) # end
+
+802.1x SSID
```
Comment (extrafu) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: 802.1X

File: lib/pf/Switch/Fortinet/FortiGate.pm [hunk 5]
```diff
@@ -10,11 +10,13 @@ The pf::Switch::Fortinet::FortiGate  module implements an object oriented interf
 
 =head1 STATUS
 
+802.1x tested with FortiOS 5.4.
```
Comment (extrafu) [inline lib/pf/Switch/Fortinet/FortiGate.pm]: 802.1X

File: lib/pf/Switch/Fortinet/FortiGate.pm [hunk 6]
```diff
@@ -10,11 +10,13 @@ The pf::Switch::Fortinet::FortiGate  module implements an object oriented interf
...
+
 =cut
 
 =head1 BUGS AND LIMITATIONS
 
-No doing deauthentication since this is a web form released switch
+No doing deauthentication in web auth
```
Comment (extrafu) [inline lib/pf/Switch/Fortinet/FortiGate.pm]: No doing? -> Not doing?

File: lib/pf/Switch/Fortinet/FortiGate.pm [hunk 7]
```diff
@@ -28,20 +30,21 @@ use LWP::UserAgent;
...
+use pf::accounting qw(node_accounting_dynauth_attr);
 
 use base ('pf::Switch::Fortinet');
 
 =head1 METHODS
 
 =cut
 
-sub description { 'FortiGate Firewall with web auth' }
+sub description { 'FortiGate Firewall with web auth + 802.1x' }
```
Comment (extrafu) [inline lib/pf/Switch/Fortinet/FortiGate.pm]: with WebAuth and 802.1X

File: lib/pf/Switch/Fortinet/FortiGate.pm [hunk 8]
```diff
@@ -158,16 +161,40 @@ sub getAcceptForm {
     return $html_form;
 }
 
-=head2 deauthenticateMacDefault
+=item deauthenticateMacDefault
 
-Just log since there is no way to deauthenticate
+Overrides base method to send Acct-Session-Id withing the RADIUS disconnect request
```
Comment (extrafu) [inline lib/pf/Switch/Fortinet/FortiGate.pm]: within

File: lib/pf/Switch/Fortinet/FortiGate.pm [hunk 9]
```diff
@@ -158,16 +161,40 @@ sub getAcceptForm {
...
+    #Fetching the acct-session-id
+    my $dynauth = node_accounting_dynauth_attr($mac);
+
+    $logger->debug("deauthenticate $mac using RADIUS Disconnect-Request deauth method");
+    return $self->radiusDisconnect(
+        $mac, { 'Acct-Session-Id' => $dynauth->{'acctsessionid'}, 'User-Name' => $dynauth->{'username'} },
+    );
 }
 
+#=head2 deauthenticateMacDefault
```
Comment (louismunro) [inline lib/pf/Switch/Fortinet/FortiGate.pm]: Why are these comments still present?
Please delete them if they are no longer used.

**Pull request merged**